### PR TITLE
Increase psalm to level 5

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because psalm is level 6 here.

### CODE REVIEW

File: tests/App/Entity/Context.php [hunk 1]
```diff
@@ -33,6 +33,6 @@ class Context extends BaseContext
 
     public function getId()
     {
-        return $this->id;
+        return $this->id ?? '';
```
Comment (VincentLanglet) [inline tests/App/Entity/Context.php:36]: Isn't better to return null if there is no id and check for null value if needed ?

Comment (jordisala1991) [reply tests/App/Entity/Context.php:36]: Yep, but that change needs to be done on SonataClassificationBundle.

Comment (VincentLanglet) [reply tests/App/Entity/Context.php:36]: Why ? What is the error you try to fix ?

Comment (jordisala1991) [reply tests/App/Entity/Context.php:36]: This is the Context entity that is created in order to test the integration with classification-bundle

(Context entity comes from classification-bundle)

Comment (VincentLanglet) [reply tests/App/Entity/Context.php:36]: I've made this https://github.com/sonata-project/SonataClassificationBundle/pull/689

Comment (VincentLanglet) [reply tests/App/Entity/Context.php:36]: It's merged @jordisala1991

Comment (jordisala1991) [reply tests/App/Entity/Context.php]: Fixed.

**Review by VincentLanglet — APPROVED** [review]

File: src/Model/NoDriverManager.php [hunk 2]
```diff
@@ -38,7 +38,7 @@ public function findAll(): array
      * @param int|null $limit
      * @param int|null $offset
      */
-    public function findBy(array $criteria, ?array $orderBy = null, $limit = null, $offset = null): array
```
Comment (franmomu) [inline src/Model/NoDriverManager.php:41]: I don't think is necessary to remove the return type

File: src/CDN/CloudFrontVersion3.php [hunk 3]
```diff
@@ -116,13 +116,22 @@ public function flushPaths(array $paths): string
...
-            $status = $result->get('Invalidation')['Status'];
+            $invalidation = $result->get('Invalidation');
+            \assert(\is_array($invalidation));
 
-            if (false === array_search($status, self::AVAILABLE_STATUSES, true)) {
-                throw new \RuntimeException(sprintf('Unable to determine the flush status from the given response: "%s".', $status));
+            $status = $invalidation['Status'] ?? null;
+
+            if (null === $status || !\in_array($status, self::AVAILABLE_STATUSES, true)) {
+                throw new \RuntimeException('Unable to determine the flush status from the given response.');
```
Comment (franmomu) [inline src/CDN/CloudFrontVersion3.php:125]: Why getting rid of the status? looks like it's useful information

Comment (jordisala1991) [reply src/CDN/CloudFrontVersion3.php]: Re add that exception (I split when having null as status from the not in the array of available statuses

File: src/Model/NoDriverManager.php (hunk 2 above)
Comment (jordisala1991) [reply src/Model/NoDriverManager.php:41]: Re added

**Review by franmomu — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**